protocols/https-2way-authz-elytron/src/test/java/org/wildfly/swarm/ts/protocols/https/twoway/authz/elytron/Https2wayAuthzElytronTest.java

```diff
@@ -0,0 +1,152 @@
+package org.wildfly.swarm.ts.protocols.https.twoway.authz.elytron;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.conn.ssl.DefaultHostnameVerifier;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContexts;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.swarm.arquillian.DefaultDeployment;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLHandshakeException;
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(Arquillian.class)
+@DefaultDeployment
+public class Https2wayAuthzElytronTest {
+    @Test
+    @RunAsClient
+    public void http() throws IOException {
+        String response = Request.Get("http://localhost:8080/").execute().returnContent().asString();
+        assertThat(response).isEqualTo("Hello on port 8080, secure: false, principal: null, isGuest: false, isUser: false");
+    }
+
+    @Test
+    @RunAsClient
+    public void https_authenticatedAndAuthorizedClient() throws IOException, GeneralSecurityException {
+        char[] clientPassword = "client-password".toCharArray();
```

Comment:
just suggestion: private static final CLIENT_PWD,
 createCustomSSLContext("keystore.jks", "truststore.jks", pwd)
 createCustomClient(context, nameVerifier, keyMaterial, trustMaterial)
These parts repeat in every test, anyway I didn't think of any better solution (builder pattern, abstract ancestor..)

Reply:
I know there's a bit of repetition here, but I think the Apache HttpClient API is so succinct it doesn't make too much sense to abstract over it again. I think I prefer to have all the code on a single place here.